backend/src/main/java/org/sefglobal/core/academix/repository/CategoryRepository.java

```diff
@@ -0,0 +1,11 @@
+package org.sefglobal.core.academix.repository;
+
+import org.sefglobal.core.academix.model.Category;
+import org.sefglobal.core.academix.projections.CustomCategory;
+import org.springframework.data.jpa.repository.JpaRepository;
+
+import java.util.List;
+
+public interface CategoryRepository extends JpaRepository<Category, Long> {
+    List<CustomCategory> findAllBy();
```

Comment:
A method already exists in `PagingAndSortingRepository` to `findAll` use that one.

Reply:
We can't use the projection if we use the existing `findAll` method.

Reply:
Have a look into Spring `Excerpts`, you should be able to achieve this by using `Excerpts`

Reply:
We cant use excerpts if we are using `@RestController`, to use `excerpts` we have to add `@RepositoryRestResource` annotation in our repository and it's incompatible with custom written `@RestController` endpoints.
- https://stackoverflow.com/questions/34393874/spring-boot-restcontroller-endpoints-and-spring-data-rest-projections
- https://stackoverflow.com/questions/33172317/spring-boot-projection-with-restcontroller